File: scratch/fix_translations.py
import json
import re

def fix_json(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        data = json.load(f)
    
    new_data = {}
    for k, v in data.items():
        # Replace spaces and special chars in keys with underscores
        new_key = re.sub(r'[^a-zA-Z0-9_]', '_', k).lower()
        # Collapse multiple underscores
        new_key = re.sub(r'_+', '_', new_key)
        # Remove leading/trailing underscores
        new_key = new_key.strip('_')
        
        if not new_key:
            new_key = "empty_key"
            
        # Handle collisions
        orig_key = new_key
        counter = 1
        while new_key in new_data:
            new_key = f"{orig_key}_{counter}"
            counter += 1
            
        new_data[new_key] = v
        
    with open(file_path, 'w', encoding='utf-8') as f:
        json.dump(new_data, f, ensure_ascii=False, indent=2)

fix_json('assets/translations/ar.json')
fix_json('assets/translations/en.json')
print("Done")

File: scratch/fix_translations.js
const fs = require('fs');

function fixJson(filePath) {
    const data = JSON.parse(fs.readFileSync(filePath, 'utf8'));
    const newData = {};
    
    for (let k in data) {
        let newKey = k.replace(/[^a-zA-Z0-9_]/g, '_').toLowerCase();
        newKey = newKey.replace(/_+/g, '_').replace(/^_|_$/g, '');
        
        if (!newKey) newKey = "empty_key";
        
        let finalKey = newKey;
        let counter = 1;
        while (newData[finalKey]) {
            finalKey = `${newKey}_${counter}`;
            counter++;
        }
        newData[finalKey] = data[k];
    }
    
    fs.writeFileSync(filePath, JSON.stringify(newData, null, 2), 'utf8');
}

fixJson('assets/translations/ar.json');
fixJson('assets/translations/en.json');
console.log("Done");
